linetools/spectra/io.py

```diff
@@ -165,26 +163,37 @@ def readspec(specfil, inflg=None, efil=None, verbose=False, multi_ivar=False,
         print('Not sure what has been input.  Send to JXP.')
         return
 
-    if np.any(np.isnan(xspec1d.dispersion)):
+
+    # Generate, as needed
+    if 'xspec1d' not in locals():
+        # Give Ang as default
+        if not hasattr(wave, 'unit'):
+            uwave = u.Quantity(wave, unit=u.AA)
+        elif wave.unit is None:
+            uwave = u.Quantity(wave, unit=u.AA)
+        else:
+            uwave = u.Quantity(wave)
+        xspec1d = XSpectrum1D.from_tuple((wave, fx, sig, None))
+
+    if np.any(np.isnan(xspec1d.wavelength)):
         warnings.warn('WARNING: Some wavelengths are NaN')
 
     # Filename
     xspec1d.filename = datfil
-    # Add in the header
-    xspec1d.head = head0
 
-    if xspec1d.co is None:
+    if not xspec1d.flag_co:
         # Final check for continuum in a separate file
-        if isinstance(specfil, basestring) and specfil.endswith('.fits'):
-            co_filename = specfil.replace('.fits', '_c.fits')
-            if os.path.exists(co_filename):
-                xspec1d.co = fits.getdata(co_filename)
-
-    if debug:
-        if xspec1d.co is None:
-            print('linetools.spectra.io.readspec(): No continuum found')
-        else:
-            print('linetools.spectra.io.readspec(): Found continuum')
+        if isinstance(specfil,basestring):
+            if specfil.endswith('.fits'):
+                try:
+                    co = fits.getdata(specfil.replace('.fits', '_c.fits'))
+                except IOError:
+                    pass
+                else:
+                    xspec1d._data['co'] = co
+
```

Comment:
Is there any reason why you're reverting to the old co file code? I changed it to avoid a try-except statement, which is slow when the exception triggers (which it usually does in this case).